src/storage/core.rs

```diff
@@ -621,6 +725,83 @@ impl Inner {
         }
     }
 
+    pub(crate) async fn restore_active_blob(&self) -> Result<bool> {
+        if self.has_active_blob().await {
+            return Ok(false);
+        }
+        let mut safe = self.safe.write().await;
+        if let None = safe.active_blob {
+            let blob_opt = safe.blobs.write().await.pop().map(|b| b.boxed());
+            if let Some(blob) = blob_opt {
+                safe.active_blob = Some(blob);
+                Ok(true)
+            } else {
+                // NOTE: maybe it would be better panic in case of this operation with empty blob list?

```

Comment:
If you think that it is highly unlikely for this case to happen, you can use expect or panic.
Either way, this case should be an Error::Unitialized